mm2src/mm2_net/src/event_streaming/sse_handler.rs

```diff
@@ -0,0 +1,79 @@
+use hyper::{body::Bytes, Body, Request, Response};
+use mm2_core::mm_ctx::MmArc;
+use serde_json::json;
+use std::convert::Infallible;
+
+pub const SSE_ENDPOINT: &str = "/event-stream";
+
+/// Handles broadcasted messages from `mm2_event_stream` continuously.
+pub async fn handle_sse(request: Request<Body>, ctx_h: u32) -> Result<Response<Body>, Infallible> {
+    let ctx = match MmArc::from_ffi_handle(ctx_h) {
+        Ok(ctx) => ctx,
+        Err(err) => return handle_internal_error(err).await,
+    };
+
+    let event_streaming_config = ctx.event_streaming_configuration();
+    if event_streaming_config.disabled {
+        return handle_internal_error("Event streaming is explicitly disabled".to_string()).await;
+    }
+
+    let client_id = match request.uri().query().and_then(|query| {
+        query
+            .split('&')
+            .find(|param| param.starts_with("id="))
+            .map(|id_param| id_param.trim_start_matches("id=").parse::<u64>())
+    }) {
+        Some(Ok(id)) => id,
+        // Default to zero when client ID isn't passed, most of the cases we will have a single user/client.
+        _ => 0,
+    };
+
+    let event_stream_manager = ctx.event_stream_manager.clone();
+    let Ok(mut rx) = event_stream_manager.new_client(client_id) else {
+        // FIXME: Such an error means that one client generated an id that is already in use
```

Comment:
I use fixmes for the PR context only (to bring my attention & others to potential issues that needs fixing). by the end of the PR dev & review there shouldn't be any fixmes left.
this avoids mangling the new short term todos with old ones, otherwise i will have to always compute the diff to find these fixmes todos while devving.